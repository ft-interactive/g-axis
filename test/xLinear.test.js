/**
 * @file
 * Test suite for xLinear
 */

import pretty from "pretty";

jest.setTimeout(20000);

beforeAll(global.build("xLinear"));
beforeEach(global.start);

test("bottom-aligned, default scales", async () => {
    await global.page.evaluate(async () => {
        const sharedConfig = {
            source: false,
            subtitle: "Bottom-aligned, default scales",
            title: "xLinear test"
        };

        const svg = await window.d3.select(document.querySelector("svg"));
        const currentFrame = window.gChartframe.webFrameMDefault(sharedConfig);

        // Adds some margin to deal with the lack of a y-axis in test
        const newMargin = currentFrame.rem();
        currentFrame.margin({ right: newMargin });

        // Set up the chart frame
        svg.call(currentFrame);

        // Instantiate xLinear
        const xAxis = window
            .xLinear()
            .range([0, currentFrame.dimension().width])
            .plotDim([
                currentFrame.dimension().width,
                currentFrame.dimension().height
            ])
            .rem(currentFrame.rem())
            .tickSize(currentFrame.rem())
            .align("bottom")
            .frameName("webFrameMDefault");

        // Set up xAxis
        currentFrame.plot().call(xAxis);

        // Translate axis to bottom of plot
        xAxis
            .xLabel()
            .attr(
                "transform",
                `translate(0,${Math.round(currentFrame.dimension().height)})`
            );
    });

    // Lame hack to deal with the unrounded numbers from d3-axis
    await global.page.evaluate(() => {
        [...document.querySelectorAll("[transform]")].map(el => {
            const transform = el.getAttribute("transform");
            const [, x, y] = transform.match(
                /translate\s?\(([\d.]+),\s?([\d.]+)\)/
            );
            if (x && y) {
                const updated = transform.replace(
                    /translate\s?\(([\d.]+),\s?([\d.]+)\)/,
                    `transform(${Math.round(x)}, ${Math.round(y)})`
                );
                el.setAttribute("transform", updated);
            }
        });
    });

    const snap = await global.page.evaluate(
        () => document.querySelector("g.chart-plot").innerHTML
    );
    expect(pretty(snap)).toMatchSnapshot();
});

test("top-aligned, default scales", async () => {
    await global.page.evaluate(async () => {
        const sharedConfig = {
            source: false,
            subtitle: "Top-aligned, default scales",
            title: "xLinear test"
        };

        const svg = await window.d3.select(document.querySelector("svg"));
        const currentFrame = window.gChartframe.webFrameMDefault(sharedConfig);

        // Adds some margin to deal with the lack of a y-axis in test
        const newMargin = currentFrame.rem();
        currentFrame.margin({ right: newMargin });

        // Set up the chart frame
        svg.call(currentFrame);

        // Instantiate xLinear
        const xAxis = window
            .xLinear()
            .plotDim([
                currentFrame.dimension().width,
                currentFrame.dimension().height
            ])
            .rem(currentFrame.rem())
            .tickSize(currentFrame.rem())
            .range([0, currentFrame.dimension().width])
            .align("top")
            .frameName("webFrameMDefault");

        // Set up xAxis
        currentFrame.plot().call(xAxis);
    });

    // Lame hack to deal with the unrounded numbers from d3-axis
    await global.page.evaluate(() => {
        [...document.querySelectorAll("[transform]")].map(el => {
            const transform = el.getAttribute("transform");
            const [, x, y] = transform.match(
                /translate\s?\(([\d.]+),\s?([\d.]+)\)/
            );
            if (x && y) {
                const updated = transform.replace(
                    /translate\s?\(([\d.]+),\s?([\d.]+)\)/,
                    `transform(${Math.round(x)}, ${Math.round(y)})`
                );
                el.setAttribute("transform", updated);
            }
        });
    });

    const snap = await global.page.evaluate(
        () => document.querySelector("g.chart-plot").innerHTML
    );
    expect(pretty(snap)).toMatchSnapshot();
});
