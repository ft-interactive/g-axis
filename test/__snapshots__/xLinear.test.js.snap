// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bottom-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis xAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\" transform=\\"transform(0, 277)\\">
  <g class=\\"tick baseline\\" opacity=\\"1\\" transform=\\"transform(0, 0)\\">
    <line stroke=\\"#000\\" y2=\\"20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"23\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">0</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(132, 0)\\">
    <line stroke=\\"#000\\" y2=\\"20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"23\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">20</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(264, 0)\\">
    <line stroke=\\"#000\\" y2=\\"20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"23\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">40</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(396, 0)\\">
    <line stroke=\\"#000\\" y2=\\"20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"23\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">60</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(528, 0)\\">
    <line stroke=\\"#000\\" y2=\\"20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"23\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">80</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(660, 0)\\">
    <line stroke=\\"#000\\" y2=\\"20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"23\\" dy=\\"0.71em\\" id=\\"webFrameMDefaultxLabel\\">100</text>
  </g>
</g>"
`;

exports[`top-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis xAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\">
  <g class=\\"tick baseline\\" opacity=\\"1\\" transform=\\"transform(0, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-23\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">0</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(132, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-23\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">20</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(264, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-23\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">40</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(396, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-23\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">60</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(528, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-23\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">80</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(660, 0)\\">
    <line stroke=\\"#000\\" y2=\\"-20\\" id=\\"webFrameMDefaultxTick\\"></line><text fill=\\"#000\\" y=\\"-23\\" dy=\\"0em\\" id=\\"webFrameMDefaultxLabel\\">100</text>
  </g>
</g>"
`;
